server.js

```diff
@@ -0,0 +1,80 @@
+import mongoose from 'mongoose';
+import cors from 'cors';
+import morgan from 'morgan'
+import Student from './models/Student.js';
+import express from 'express';
+import './db-connect.js';
+
+const app = express();
+
+app.use(express.json()); //  parses incoming JSON into req.body
+app.use(cors()); // allow access to our Api from front end
+app.use(morgan("dev")); // logs all my accesses to routes to the terminal
+
+app.get("/students", async (req, res)=> {
+    const students = await Student.find()
+    res.json(students);
+
+}) 
+
+app.get("/students/migrate", async(req, res)=> {
+
+    const updated = await Student.updateMany().exists("role", false).set("role", "user")
+
+    res.json(updated);
+})
+
+app.get('/students/:id', async (req, res, next)=> {
+    const {id} = req.params
+    try {
+        const student = await Student.findById(id)
+        res.json(student.firstname)
+    } catch(err) {
+        next(err)
+    }
+})
+
+app.post('/students', async (req, res, next)=> {
+    try {
+        const body = req.body
+        const newbie = await Student.create(body)
+        res.json(newbie)
+    } catch(err) {
+        next(err)
+    }
+})
+
+app.put('/students/:id', async (req, res, next)=> {
+    try {
+        const {id} = req.params;
+        const studentData = req.body;
+        const studentUpdated = await Student.findByIdAndUpdate(id, studentData, {new:true});
+        res.json(studentUpdated)
+    } catch(err) {
+        next(err)
+    }
+})
+
+app.delete("/students/:id", async(req, res, next)=> {
+    try {
+        const {id} = req.params;
+        const studentDeleted = await Student.findByIdAndDelete(id);
+        res.json({done: `ID ${id} is deleted`});
```

Comment:
Here you can actually return the deleted student, so `res.json( studentDeleted )`
Even though it is a bit confusing, findByIdAndDelete() will actually deliver back the deleted student. And the deleted item we would return back to the frontend, as a convention. The frontend can then decide what to do with it...